agnostic-svelte/src/lib/components/Input/Input.svelte

```diff
@@ -1,391 +1,409 @@
-  export let type = 'text';
-
-  $: if (!value) value = "";
-  $: inputType = type;
-
-  $: labelClasses = [
-    "label",
-    isInvalid ? "label-error" : "",
-    isInline ? "label-inline" : "",
-    size ? `label-${size}` : "",
-    labelCss ? labelCss : "",
-  ].filter(c => c).join(" ");
-
-  $: inputClasses = [
-    isSkinned ? "input" : "input-base",
-    isRounded ? "input-rounded" : "",
-    isUnderlined ? "input-underlined" : "",
-    hasLeftAddon ? "input-has-left-addon" : "",
-    hasRightAddon ? "input-has-right-addon" : "",
-    isDisabled ? "disabled" : "",
-    isInvalid ? "input-error" : "",
-    isInline ? "input-inline" : "",
-    isUnderlinedWithBackground ? "input-underlined-bg" : "",
-    css ? css : "",
-    size ? `input-${size}` : "",
-  ].filter(c => c).join(" ");
-
-  $: invalidClasses = () => {
-    return size ? `field-error-${size}` : "field-error";
-  };
-  
-  $: helpClasses = () => {
-    return size ? `field-help-${size}` : "field-help";
-  };
-  
-  $: addonContainerClasses = () => "input-addon-container";
-</script>
-<div class="w-100">
-  <label class={labelClasses} for={id}>{label}</label>
-  {#if type == "textarea"}
-    <textarea
-      id={id}
-      class={inputClasses}
-      on:blur
-      on:change
-      bind:value
-      on:click
-      on:focus
-      {...$$restProps}></textarea>
-  {:else if hasLeftAddon || hasRightAddon}
-    <div class={addonContainerClasses()}>
-      <slot name="addonLeft" />
+    background: var(--agnostic-input-disabled-bg, var(--agnostic-disabled-bg)) !important;
+    color: var(--agnostic-input-disabled-color, var(--agnostic-disabled-color)) !important;
+    appearance: none !important;
+    box-shadow: none !important;
+    cursor: not-allowed !important;
+    opacity: 80% !important;
+  }
+  
+  @media screen and (-ms-high-contrast: active) {
+    /* High contrast mode outline hacks */
+  
+    /* styleint-disable-next-line no-descending-specificity  */
+    .input:disabled {
+      outline: 2px solid transparent;
+      outline-offset: -2px;
+    }
+  }
+  
+  /**
+   * Input "has addon"
+   */
+  
+  .input-addon-container {
+    display: flex;
+    position: relative;
+    width: 100%;
+  }
+  
+  .input-has-left-addon {
+    padding-left: calc(var(--agnostic-side-padding) * 3);
+  }
+  
+  .input-has-right-addon {
+    padding-right: calc(var(--agnostic-side-padding) * 3);
+  }
+  
+  .input-addon-left {
+    left: var(--agnostic-input-side-padding);
+  }
+  
+  .input-addon-right {
+    right: var(--agnostic-input-side-padding);
+  }
+  
+  @media (prefers-reduced-motion), (update: slow) {
+    /* stylelint-disable selector-no-vendor-prefix */
+    .input-skin,
+    .input,
+    .input::placeholder,
+    .input::-webkit-input-placeholder,
+    .input::-ms-placeholder,
+    .input:-ms-placeholder,
+    .input:focus {
+      transition-duration: 0.001ms !important;
+    }
+  }
+  
+  </style>
+  
+  <script lang="ts">
```

Comment:
Yikes, this is my fault 🤦‍♂️ https://github.com/AgnosticUI/agnosticui/blob/24483d0b8651888a9b2362bff85020f5c775c1b3/agnostic-svelte/src/lib/components/Input/Input.svelte#L312 -- I'll remove this later. We ARE porting to typescript in agnostic-svelte-ts/* but this is a mistake on my end.